### PR TITLE
Rewrite navbar generation code from first principles

### PR DESCRIPTION
**Pull request by hadley**
@maelle I got tired of having a bunch of code copied from rmarkdown and then modified over the years. It was hard to understand the logic and challenging to make changes. So I've rewritten it from scratch. Along the way:

* I documented the `target` field for links
* I discovered that we were using `depth` to mean two different things (menu and path), causing search to fail pages deeper in the site
* I updated the generated HTML to match latest BS5 recommendations (i.e. using `<ul>` and `<li>` instead of bare `<div>`s
* I learned that BS5 definitely doesn't support submenus and now give a slighter better error message (instead of just generating HTML that looks weird)
* I added some better errors if your navbar code is malformed

### CODE REVIEW

**Comment by jayhesselberth** [comment]
This looks good but the position of the search box is slightly different on the new site — is there a CSS rule that also needs updating?

**Review by maelle — COMMENTED** [review]
Nice! 

- should there be tests for the two other new errors?
- is it worth re-running an HTML validator?
- should there a "gallery" of some YAML configurations for the navbar in the tests?

File: R/navbar-menu.R [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  )
+
+  html_tag("form", class = "form-inline", role = "search", "\n", input, "\n")
+}
+
+# Reused HTML components -----------------------------------------------------
+
+html_tag <- function(tag, ..., class = NULL) {
+  dots <- list2(...)
+  dots_attr <- dots[names2(dots) != ""]
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/navbar-menu.R:161]: I always forget about that function because the grammar is a bit tortured 😄

**Comment by hadley** [comment]
* Yes, done!
* Already did it 😄 
* I think the existing tests are ok since the structure is now pretty simple given that we are clear are not about supporting submenus. But I'll add a comment to explain a bit more.

**Pull request merged**